nat/src/stateless/mod.rs

```diff
@@ -86,53 +80,84 @@ impl StatelessNat {
         }
     }
 
-    fn translate_src(net: &mut Net, ranges_src_nat: &TrieValue) -> Option<()> {
-        let current_src_ip = net.src_addr();
-        let target_src_ip = map_ip_src_nat(ranges_src_nat, &current_src_ip);
-        match (net, target_src_ip) {
-            (Net::Ipv4(hdr), IpAddr::V4(src_ip)) => {
-                debug!("Changing ipv4 src: {current_src_ip} -> {src_ip}");
-                hdr.set_source(UnicastIpv4Addr::new(src_ip).ok()?);
+    /// Get the name of this instance
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+
+    /// Translate packet source ip address.
+    /// # Errors
+    /// Returns `NatError::UnsupportedTranslation` if the translation is unsupported. On success, returns `Ok` indicating
+    /// if the address did actually change or not, since the NAT module may map it to the same address.
+    fn translate_src(&self, net: &mut Net, ranges_src_nat: &TrieValue) -> Result<bool, NatError> {
+        let nfi = self.name();
+        let current_src = net.src_addr();
+        let target_src = map_ip_src_nat(ranges_src_nat, &current_src);
+        if target_src == current_src {
+            return Ok(false);
+        }
+        match (net, target_src) {
+            (Net::Ipv4(hdr), IpAddr::V4(src)) => {
+                debug!("{nfi}: Changing ipv4 src: {current_src} -> {src}");
```

Comment:
Same, I'm not a fan of starting a log line with a non-deterministic name. I can clean this up later, though.

Reply:
Why? That's the name of the stage.

Reply:
The person who writes the pipeline stage can use whatever name they want, right? They can pick something that looks like `StatelessNAT01` which is explicit, or just as well something like `demo-next-month-xlate` which may or may not be intuitive when parsing the logs. Prefixing with `StatelessNat` would make it obvious what sort of stage we're talking about, however it is named.